src/System.Console/tests/Color.cs

```diff
@@ -92,6 +92,68 @@ public static void NonRedirectedOutputDoesUseAnsiSequences()
         Assert.Equal(3, Encoding.UTF8.GetString(data.ToArray()).ToCharArray().Count(c => c == Esc));
     }
 
+    [Fact]
+    [PlatformSpecific(PlatformID.AnyUnix)]
+    public void VerifyInstalledTermInfosParse()
+    {
+        foreach (string location in ConsolePal.TermInfo.Database._terminfoLocations)
+        {
+            // Skip directories that dont exist
+            if (Directory.Exists(location) == false)
+                continue;
+
+            foreach (string term in Directory.EnumerateFiles(location, "*", SearchOption.AllDirectories))
+            {
+                if (term.ToUpper().Contains("README")) continue;
+                ConsolePal.TermInfo.Database db = ConsolePal.TermInfo.Database.ReadDatabase(Path.GetFileName(term));
+                Assert.True(db != null, string.Format("Could not parse terminfo file {0}", term)); // Don't use Assert.NotNull so we can print out the term info file on failure
+            }
+        }
+    }
+
+    [Fact]
+    [PlatformSpecific(PlatformID.AnyUnix)]
+    public void VerifyXterm256ColorTermInfoParsesCorrectly()
+    {
+        TermInfoVerification("xterm-256color", "\u001B\u005B\u0025\u003f\u0025p1%{8}%<%t3%p1%d%e%p1%{16}%<%t9%p1%{8}%-%d%e38;5;%p1%d%;m", "\u001B\u005B\u0025\u003f\u0025p1%{8}%<%t4%p1%d%e%p1%{16}%<%t10%p1%{8}%-%d%e48;5;%p1%d%;m");
+    }
+
+    [Fact]
+    [PlatformSpecific(PlatformID.AnyUnix)]
+    public void VerifyScreenTermInfoParsesCorrectly()
+    {
+        TermInfoVerification("screen", "\u001B\u005B\u0033\u0025p1%dm", "\u001B\u005B\u0034\u0025p1%dm");
+    }
+
+    [Fact]
+    [PlatformSpecific(PlatformID.AnyUnix)]
+    public void VerifyEtermTermInfoParsesCorrectly()
+    {
+        TermInfoVerification("Eterm", "\u001B\u005B\u0033\u0025p1%dm", "\u001B\u005B\u0034\u0025p1%dm");
+    }
+
+    [Fact]
+    [PlatformSpecific(PlatformID.AnyUnix)]
+    public void VerifyWsvt25TermInfoParsesCorrectly()
+    {
+        TermInfoVerification("wsvt25", "\u001B\u005B\u0033\u0025p1%dm", "\u001B\u005B\u0034\u0025p1%dm");
+    }
+
+    [Fact]
+    [PlatformSpecific(PlatformID.AnyUnix)]
+    public void VerifyMachColorTermInfoParsesCorrectly()
+    {
+        TermInfoVerification("mach-color", "\u001B\u005B\u0033\u0025p1%dm", "\u001B\u005B\u0034\u0025p1%dm");
+    }
+
+    // We need to compare bytes since the strings output to the console are different than the underlying bytes.
+    private void TermInfoVerification(string termToTest, string expectedForeground, string expectedBackground)
+    {
+        ConsolePal.TerminalColorInfo info = new ConsolePal.TerminalColorInfo(ConsolePal.TermInfo.Database.ReadDatabase(termToTest));
```

Comment:
A few things:
1. What if this termToTest doesn't exist on the system on which the test is running?
2. The strings being passed in here are only part of what needs to be tested.  These are the strings that are pulled out of the terminfo file, but they're not what's written out to stdout.  Rather, these strings are actually an interpreted language that we need to evaluate with inputs (in the case of Foreground/BackgroundColor, the numeric values of the selected colors) in order to determine what strings to actually write out to the console: https://github.com/dotnet/corefx/blob/master/src/System.Console/src/System/ConsolePal.Unix.cs#L199-L201 .  We should be doing some level of verification on that evaluation process, at the very least to ensure we don't throw exceptions, but ideally something beyond that.  I suggest that in these tests for known terminfos, you verify the output that gets generated for a variety of color values, and in the overall test that parses all of the terminfo files, you do more basic verification, e.g. if the foreground/background strings exist, that running them through the evaluator doesn't throw and produces a non-empty string.